pkg/dbutil/retry.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dbutil
+
+import (
+	"strings"
+
+	"github.com/go-sql-driver/mysql"
+	"github.com/pingcap/errors"
+	tmysql "github.com/pingcap/parser/mysql"
+)
+
+var (
+	// Retryable1105Msgs list the error messages of some retryable error with `1105` code (`ErrUnknown`).
+	Retryable1105Msgs = []string{
+		"Information schema is out of date",
+		"Information schema is changed",
+	}
+)
+
+// IsRetryableError checks whether the SQL statement can be retry directly when encountering this error.
+// NOTE: this should be compatible with different TiDB versions.
+// some errors are only retryable in some special cases, then we mark it as un-retryable:
+// - tmysql.ErrTiKVServerTimeout
+// - tmysql.ErrWriteConflictInTiDB
+// - tmysql.ErrTableLocked
+// - tmysql.ErrWriteConflict
+//
+// some errors are un-retryable:
+// - tmysql.ErrQueryInterrupted
+//
+// some errors are unknown:
+// - tmysql.ErrRegionUnavailable
+func IsRetryableError(err error) bool {
+	err = errors.Cause(err) // check the original error
+	mysqlErr, ok := err.(*mysql.MySQLError)
+	if !ok {
+		return false
+	}
+
+	switch mysqlErr.Number {
+	case tmysql.ErrLockDeadlock: // https://dev.mysql.com/doc/refman/5.7/en/innodb-deadlocks.html
+		return true // retryable error in MySQL
+	case tmysql.ErrPDServerTimeout,
```

Comment:
seems DM had more retryable error，will DM use this pkg later?

Reply:
Yes, DM will use this pkg. But some retryable errors in DM are not actual retryable, please see the comment of this function.

Reply:
ok